docs/backend-system/building-backends/01-index.md

```diff
@@ -10,3 +10,169 @@ description: Building backends using the new backend system
+
+A more advanced way to deploy Backstage is to split the backend plugins into multiple different backend deployments. Both the [deployment documentation](../../deployment/scaling.md) and [Threat Model](../../overview/threat-model.md#trust-model) explain the benefits of this, so here we'll focus on how to do it.
+
+To create a separate backend we need to create an additional backend package. This package will be built and deployed separately from your existing backend. There is currently no template to create a backend via `yarn new`, so the quickest way is to copy the new package and modify. The naming is up to you and it depends on how you are splitting things and up, for this example we'll just use a simple suffix. You might end up with a directory structure like this:
+
+```text
+packages/
+  backend-a/
+    src/
+      index.ts
+    package.json <- "name": "backend-a"
+  backend-b/
+    src/
+      index.ts
+    package.json <- "name": "backend-a"
+```
+
+You can now trim down the `src/index.ts` files to only include the plugins and modules that you want to be part of that backend. For example, if you want to split out the scaffolder plugin, you might end up with something like this:
+
+```ts
+// packages/backend-a/src/index.ts, imports omitted
+const backend = createBackend();
+
+backend.add(appPlugin());
+backend.add(catalogPlugin());
+backend.add(catalogModuleTemplateKind());
+backend.start();
+```
+
+And `backend-b`, don't forget to clean up dependencies in `package.json` as well:
+
+```ts
+// packages/backend-b/src/index.ts, imports omitted
+const backend = createBackend();
+
+backend.add(scaffolderPlugin());
+backend.start();
+```
+
+We've now split the backend into two separate deployments, but we still need to make sure that they can communicate with each other. This is the hard and somewhat tedious part, as Backstage currently doesn't provide and out of the box solution that solves this. You'll need to manually configure the two backends with custom implementations of the `DiscoveryService` and have them return the correct URLs for each other. Likewise, you'll also need to provide a custom implementation of the `DiscoveryApi` in the frontend, unless you surface the two backends via a proxy that handles the routing instead.
+
+### Shared Environments
+
+To make it a bit easier to manage multiple backends, it's possible to create a shared environment that can be used across multiple backends. You would typically house it in a separate package that can be referenced by backends in your monorepo, or published to a package registry for broader use.
+
+A shared environment contains a set of service implementations that should be used across all backends. These services will override the default ones, but if a service is provided directly to the backend, it will override the one in the shared environment.
+
+A shared environment is defined using `createSharedEnvironment`:
+
+```ts
+// packages/backend-env/src/index.ts
```

Comment:
To appease eager copy-pasters, it might be wise to have a separate paragraph explicitly saying essentially that "In this example, we have chosen to add the environment to the package `packages/backend-env` which likely does not exist in your installation. You can create it using `<yarn new command with the right role selected>`"

Reply:
Yeah that `yarn new` command doesn't exist at the minute :p

I'm leaning into the advanced use-case here, basically expecting anyone that attempts this to know to create the package etc. I think in the future as this is easier to do we should for sure update this section.